pkg/installation/install.go

```diff
@@ -143,12 +147,20 @@ func createOrUpdateLink(binDir string, binary string, plugin string) error {
 	return nil
 }
 
-func removeLink(binDir string, plugin string) error {
-	dst := filepath.Join(binDir, pluginNameToBin(plugin))
-	if err := os.Remove(dst); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("failed to remove the symlink in %q, err: %v", dst, err)
-	} else if err == nil {
-		glog.V(3).Infof("Removed symlink from %q", dst)
+// removeLink removes a symlink reference if exists.
+func removeLink(path string) error {
```

Comment:
I don't think we need a generic remove link function. Can we keep the old function interface? `removeLink(binDir string, plugin string)`, that makes it easier to use.

Reply:
@lbb That is really not clean code and vastly reduces testability.

We call `filepath.Join(bin, pluginNameToBin(name))` in so many unrelated places  it's at it is pointing to code smell.